tidy: replace `make check` with `./x.py test` in documentation

chrissimpkins opened the pull request:
This PR includes a minor documentation update for tidy.  It replaces the `make check` approach with `./x.py test` and describes how to execute the tidy checks (only) with ~~`./x.py test src/tools/tidy`~~ `./x.py test tidy`.

rust-highfive commented:
r? @nikomatsakis

(rust_highfive has picked a reviewer for you, use r? to override)

petrochenkov commented on src/tools/tidy/src/main.rs:
```diff
@@ -1,8 +1,8 @@
-//! etc. This is run by default on `make check` and as part of the auto
-//! builders.
+//! etc. This is run by default on `./x.py test` and as part of the auto
+//! builders. The tidy checks can be executed with `./x.py test src/tools/tidy`.
```
There's also a shortcut `./x.py test tidy`.

petrochenkov replied:
(And `./x.py test toolname` in general.)

chrissimpkins replied:
> There's also a shortcut ./x.py test tidy

Ah! Very nice.  I don't see this in the `./x.py --help` or Rustc guide.  Is this the preferred approach?  Happy to add this to both of those locations and update here if so.

> (And ./x.py test toolname in general.)

Great!

petrochenkov replied:
>Is this the preferred approach?

Well, it's my preferred approach at least, because it's shorter to type.
I don't know about others.

chrissimpkins commented:
re: https://github.com/rust-lang/rust/pull/69603#discussion_r386092523

@petrochenkov Here is what I see with `./x.py tidy -h`.  I think that we should fix this if we are going to consider that sub-command the default approach.

```
info: Downloading and building bootstrap before processing --help
      command. See src/bootstrap/README.md for help with common
      commands.
Updating only changed submodules
Submodules updated in 0.05 seconds
    Finished dev [unoptimized] target(s) in 0.15s
Usage: x.py <subcommand> [options] [<paths>...]

Subcommands:
    build       Compile either the compiler or libraries
    check       Compile either the compiler or libraries, using cargo check
    clippy      Run clippy (uses rustup/cargo-installed clippy binary)
    fix         Run cargo fix
    fmt         Run rustfmt
    test        Build and run some test suites
    bench       Build and run some benchmarks
    doc         Build documentation
    clean       Clean out build directories
    dist        Build distribution artifacts
    install     Install distribution artifacts

To learn more about a subcommand, run `./x.py <subcommand> -h`

failed to run: /Users/chris/code-ext/rustlang/rust/build/bootstrap/debug/bootstrap tidy -h
```

nikomatsakis commented:
r? @petrochenkov -- I didn't know about the shortcut either, so clearly I'm not the best to review. :)

petrochenkov commented:
>Here is what I see with `./x.py tidy -h`.

This is not surprising, given that `tidy` is not a subcommand by itself, but a target of the `x.py test` command.

petrochenkov commented:
`x.py test --help` could say something about `x.py test src/tools/tool` and the `x.py test tool` shortcut.

chrissimpkins commented:
Thanks Vadim! 

> This is not surprising, given that tidy is not a subcommand by itself, but a target of the x.py test command.

Apologies.  I think that I misread the earlier comment.

> x.py test --help could say something about x.py test src/tools/tool and the x.py test tool shortcut.

 I will update the test help and let you know when it is available.

chrissimpkins commented:
> There's also a shortcut ./x.py test tidy.

Added in https://github.com/rust-lang/rust/pull/69603/commits/0982c58edc5b9fe4cfc8b22bc3edaa95e4f5c638 https://github.com/rust-lang/rust/pull/69603/commits/259b06e33f533096f0b46beca8719fd4afceb78b

> x.py test --help could say something about x.py test src/tools/tool and the x.py test tool shortcut.

Added in https://github.com/rust-lang/rust/pull/69603/commits/9f734c978c4c32eea552d45fef582804c1d2b9bf

The `./x.py test --help` includes the following information after these changes:

<img width="940" alt="2020-03-09_22-05-50" src="https://user-images.githubusercontent.com/4249591/76272760-91223580-6252-11ea-8740-f60e3d63cbe5.png">

I believe that I addressed all of your suggestions.  Please let me know if there are any other locations in this repository where this information might be of use.

chrissimpkins commented:
https://github.com/rust-lang/rustc-guide/pull/607 adds the same recommendation to the `rustc-dev-guide`

petrochenkov commented:
@bors r+ rollup

bors commented:
:pushpin: Commit c60d581eb15d6fbe17a657da7c1de4dc82b70b68 has been approved by `petrochenkov`

<!-- @bors r=petrochenkov c60d581eb15d6fbe17a657da7c1de4dc82b70b68 -->
<!-- homu: {"type":"Approved","sha":"c60d581eb15d6fbe17a657da7c1de4dc82b70b68","approver":"petrochenkov"} -->

bors commented:
:evergreen_tree: The tree is currently closed for pull requests below priority 1000, this pull request will be tested once the tree is reopened
